libqtile/command_client.py

```diff
@@ -220,23 +220,42 @@ def __getitem__(self, name: str) -> "InteractiveCommandClient":
         Parameters
         ----------
         name : str
-            The name of the item to resolve
+            The name, or index if it's of int type, of the item to resolve
 
         Return
         ------
         InteractiveCommandClient
             The current client, navigated to the specified command graph
             object.
         """
+        if isinstance(self._current_node, CommandGraphRoot):
+            raise KeyError("Root node has no available items",
+                           name, self._current_node.selectors)
+
         if not isinstance(self._current_node, CommandGraphObject):
-            raise SelectError("Unable to make selection on current node", name, self._current_node.selectors)
+            raise SelectError("Unable to make selection on current node",
+                              name, self._current_node.selectors)
 
         if self._current_node.selector is not None:
-            raise SelectError("Selection already made", name, self._current_node.selectors)
-
-        # check that the selection is valid
-        if not self._command.has_item(self._current_node.parent, self._current_node.object_type, name):
-            raise SelectError("Item not available in object", name, self._current_node.selectors)
+            raise SelectError("Selection already made", name,
+                              self._current_node.selectors)
+
+        def _check_item(item):
+            """check that the selection is valid in the server-side qtile manager"""
+            if not self._command.has_item(self._current_node.parent,
+                                          self._current_node.object_type, item):
+                raise SelectError("Item not available in object",
+                                  str(item), self._current_node.selectors)
+
+        if isinstance(name, str) and name.isdigit():
+            # Check the item as is, and check its int version once more if it fails
+            try:
+                _check_item(name)
+            except SelectError:
+                name = int(name)
+                _check_item(name)
```

Comment:
I really don't like this. We should have some way to know whether we must send a string or int.

Reply:
> Also related to #1298 right?

Yeah!

How about normalizing the item at first? There is no way to determine if an item is int or str from shell arguments.

Reply:
From shell arguments no, but from functions parameters yes!

Reply:
Now it's normalized in `qtile_cmd.py`, only where it's necessary.